mmv1/products/tpuv2/QueuedResource.yaml

```diff
@@ -0,0 +1,110 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'QueuedResource'
+description: |
+  A Cloud TPU Queued Resource.
+min_version: 'beta'
+references:
+  guides:
+    'Official Documentation': 'https://cloud.google.com/tpu/docs/'
+  api: 'https://cloud.google.com/tpu/docs/reference/rest/v2/projects.locations.queuedResources'
+base_url: 'projects/{{project}}/locations/{{zone}}/queuedResources'
+self_link: 'projects/{{project}}/locations/{{zone}}/queuedResources/{{name}}'
+create_url: 'projects/{{project}}/locations/{{zone}}/queuedResources?queuedResourceId={{name}}'
+update_url: 'projects/{{project}}/locations/{{zone}}/queuedResources/{{name}}'
+update_verb: 'PATCH'
+update_mask: true
+autogen_async: true
+async:
+  actions: ['create', 'delete', 'update']
```

Comment:
Can this resource be updated at all? The [documentation](https://cloud.google.com/tpu/docs/reference/rest/v2/projects.locations.queuedResources) doesn't show an update methods.

If the overall resource is immutable, we can mark it as immutable by setting [immutable:true](https://googlecloudplatform.github.io/magic-modules/develop/resource/#add-a-resource:~:text=%23-,immutable%3A%20true,-%23%20URL%20for%20the) as a top-level property in this yaml. At the same time it'd be good to remove update from the list of async actions.

Reply:
Removed 'update'. I don't think the overall resource is immutable because the `state` output field among other nested fields can be changed.

Reply:
When [resources are marked as immutable in the YAML](https://googlecloudplatform.github.io/magic-modules/develop/resource-reference/#immutable) it causes all the fields to be marked as [ForceNew](https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#forcenew). This means if a user changes any user-supplied argument in the resource block for the `google_tpu_v2_queued_resource` resource then the resource will be recreated (delete+create) instead of updated in place. Any changes to output values that come from the API only and aren't user supplied values will still be updated when Terraform performs refresh action.

Does that make sense? I think marking the resource as immutable is necessary due to the lack of an update method in the API but this would not prevent the API changing output values and users being able to see those changed output values in their plans and state.

Reply:
When manually testing updating a google_tpu_v2_queued_resource resource as it's currently implemented (immutable:false) I see this:

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # google_tpu_v2_queued_resource.qr will be updated in-place
  ~ resource "google_tpu_v2_queued_resource" "qr" {
        id      = "projects/redacted/locations/us-central1-c/queuedResources/tf-test-test-qr"
        name    = "tf-test-test-qr"
        # (2 unchanged attributes hidden)

      ~ tpu {
          ~ node_spec {
                # (2 unchanged attributes hidden)

              ~ node {
                  ~ description      = "Text description of the TPU." -> "UPDATED Text description of the TPU."
                    # (2 unchanged attributes hidden)
                }
            }
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_tpu_v2_queued_resource.qr: Modifying... [id=projects/redacted/locations/us-central1-c/queuedResources/tf-test-test-qr]
╷
│ Error: Error updating QueuedResource "projects/redacted/locations/us-central1-c/queuedResources/tf-test-test-qr": googleapi: got HTTP response code 404 with body: <!DOCTYPE html>
```

Reply:
Updated the resource to be immutable.

That makes sense if it's only the user-supplied fields which cannot be changed and not the output values, and makes sense since there is no update method.